modules/sdp/SdpSimulcast.ts

```diff
@@ -0,0 +1,307 @@
+                value: newCname
+            });
+        }
+
+        mLine.ssrcGroups.push({
+            semantics: 'SIM',
+            ssrcs: cachedSsrcs.join(' ')
+        });
+
+        return mLine;
+    }
+
+    /**
+     * Generates a new set of SSRCs for the higher simulcast layers/streams and adds the attributes and SIM group to
+     * the given media description and returns the modified media description.
+     *
+     * @param mLine
+     * @param primarySsrc
+     * @returns
+     */
+    _generateNewSsrcsForSimulcast(mLine: transform.MediaDescription, primarySsrc: number) : any {
+        const cname = this._getSsrcAttribute(mLine, primarySsrc, 'cname');
+        let msid = this._getSsrcAttribute(mLine, primarySsrc, 'msid');
+        const addAssociatedAttributes = (mLine: transform.MediaDescription, ssrc: number) => {
+            mLine.ssrcs.push({
+                id: ssrc,
+                attribute: "cname",
+                value: cname
+            });
+            mLine.ssrcs.push({
+                id: ssrc,
+                attribute: "msid",
+                value: msid
+            });
+        }
+
+        // In Unified-plan mode, the a=ssrc lines with the msid attribute are not present (only cname attributes are
+        // present) in the answers that Chrome and Safari generate for an offer received from Jicofo. Generate these
+        // a=ssrc lines using the msid values from the a=msid line.
+        if (!msid) {
+            msid = mLine.msid;
+            const primarySsrcs = mLine.ssrcs;
+
+            primarySsrcs.forEach(ssrc => {
+                mLine.ssrcs.push({
+                    id: ssrc.id,
+                    attribute: "msid",
+                    value: msid
+                });
+            })
+        }
+
+        // Generate SIM layers.
+        const simSsrcs = [];
+    
+        for (let i = 0; i < this._options.numOfLayers - 1; ++i) {
+            const simSsrc = this._generateSsrc();
+
+            addAssociatedAttributes(mLine, simSsrc);
+            simSsrcs.push(simSsrc);
+        }
+
+        mLine.ssrcGroups = mLine.ssrcGroups || [];
+        mLine.ssrcGroups.push({
+            semantics: "SIM",
+            ssrcs: primarySsrc + " " + simSsrcs.join(" ")
+        });
+    
+        return mLine;
+    }
+
+    /**
+     * Returns a random number to be used for the SSRC.
+     *
+     * @returns
+     */
+    _generateSsrc() : number {
+        const min = 0, max = 0xffffffff;
+
+        return Math.floor(Math.random() * (max - min)) + min;
+    }
+
+    /**
+     * Returns the requested attribute value for a SSRC from a given media description.
+     *
+     * @param mLine
+     * @param ssrc
+     * @param attributeName
+     * @returns
+     */
+    _getSsrcAttribute(mLine: transform.MediaDescription, ssrc: number, attributeName: string) : string {
+        return mLine.ssrcs
+            .filter(ssrcInfo => parseInt(ssrcInfo.id?.toString()) === ssrc)
```

Comment:
ssrcInfo.id is of type `string | number ` and therefore coercion is necessary, otherwise TS complains.

Reply:
can ssrc's id technically be anything else than a number? maybe we should adjust the sdp-transform library?

Reply:
I am reading the types from [this](https://github.com/clux/sdp-transform) library. Its ok to go with the type coercion here instead of spending to submit a PR there.

Reply:
What do you mean by type coercion in this case exactly? I'm only worried that parsing is not really a type coercion, but it has performance implications. I'd rather put a ts ignore here or do a cast (you can write `as number` I think).